src/test/java/io/inholland/groep4/api/service/AccountsServiceTest.java

```diff
@@ -0,0 +1,159 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.BOOLEAN;
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+public class AccountsServiceTest {
+
+    @Autowired
+    private UserAccountService userAccountService;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    public JwtTokenProvider jwtTokenProvider;
+
+    @Test
+    public void gettingAllAccountsShouldGiveListOfUsers() {
+        List<UserAccount> accounts = userAccountService.getAllAccounts();
+
+        assertThat(accounts).isNotEmpty();
+        assertThat(accounts.get(0).getAccountBalance()).isEqualTo(500.0);
+    }
+    @Test
+    public void gettingAccountsByIdShouldGiveAccount() {
+        Long idToFind = 1l;
+        UserAccount account = userAccountService.getAccountById(idToFind);
+
+        assertThat(account).isNotNull();
+        assertThat(account.getAccountBalance()).isEqualTo(500.0);
+    }
+
+    @Test
+    public void GettingAccountWithWrongIDShouldThrowException(){
+        Long idToFind = 100l;
+
+        Exception exception = assertThrows(ResponseStatusException.class, () -> {
+            userAccountService.getAccountById(idToFind);
+        });
+
+        assertTrue(exception.getMessage().contains("Id not found"));
+    }
+
+    @Test
+    public void gettingAccountByUserShouldGiveUserAccount(){
+        String usernameToFind = "test-user1";
+        User username = userService.findByUsername(usernameToFind);
+        List<UserAccount> userAccount = userAccountService.getAccountsByUser(username);
+
+        assertThat(userAccount).isNotEmpty();
+        assertThat(username.getUsername()).isEqualTo("test-user1");
+    }
+    @Test
+    public void AddIncorrectNewIBAN(){
+        User user = new User();
+        user.setUsername("peter");
+        user.setPassword("test");
+        user.setFirstName("Peter");
+        user.setLastName("Griffin");
+        user.setEmail("peter@example.com");
+        user.setBirthdate("01/01/1970");
+
+        UserAccount userAccount = new UserAccount();
+        userAccount.setAccountType(UserAccount.AccountTypeEnum.CURRENT);
+        userAccount.setOwner(user);
+        userAccount.setAccountBalance(0.00);
+        userAccount.setAccountStatus(UserAccount.AccountStatusEnum.ACTIVE);
+        userAccount.setLowerLimit(100.00);
+        userAccount.setIBAN("NL01INHO0000000001");
+
+        String IBAN = userAccountService.getIBAN();
+        boolean ExistByIBAN = userAccountService.existByIBAN(IBAN);
+        UserAccount newUser = userAccountService.add(userAccount, ExistByIBAN);
+
+        assertThat(newUser).isNotNull();
```

Comment:
Als het goed is zou er een foutmelding moeten komen dat de IBAN al in gebruik is.
Alleen doe je hier vervolgens niets mee. De assert die je doet heeft niets te maken met een UserAccount.